docker/Dockerfile

```diff
@@ -138,6 +137,25 @@ ENV PATH="${PATH}:/usr/local/lib/nodejs/node-v17.9.0-linux-x64/bin"
 
 RUN npm install --global markdownlint-cli@0.31.1
 
+FROM dependencies as dependencies-dev-platform-arm64
```

Comment:
These had to be split because I couldn't do `https://github.com/hadolint/hadolint/releases/download/v2.10.0/hadolint-Linux-$TARGETARCH`.

`TARGETARCH` comes back `amd64` in our normal workflow, but hadolint doesn't publish a release named that way.

Reply:
The other option here is to move these steps into a script which can do platform detection and run according.

Reply:
Was going to ask if there was a way to collapse in one stage, would rather not do a build script inside a dockerfile.

Reply:
@whitej6 better now?

Reply:
yes